### PR TITLE
Sentry Logging for Ontology

### PR DESCRIPTION
**Pull request by rushtong**
Addresses: https://broadinstitute.atlassian.net/browse/GAWB-2116

Also see: https://github.com/broadinstitute/consent/pull/246
Requires: https://github.com/broadinstitute/firecloud-develop/pull/699

Log to sentry via the raven plugin. 
Dockerfile changes are to better align with both FireCloud and Consent.

Requires config changes in firecloud-develop for the non-local/fiab environments
```
logging:
  level: INFO
  loggers:
    appenders:
      - type: console
        threshold: INFO
        target: stdout
      - type: raven
        threshold: WARN
        dsn: ... dsn from vault ... 
        environment: ... env from vault ...
```
Also requires a change to the app startup so it gets the dsn injected: `-Dsentry.dsn=<pull secret from vault>`

---

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/java/org/broadinstitute/dsde/consent/ontology/OntologyApp.java [hunk 1]
```diff
@@ -28,6 +28,11 @@
 public class OntologyApp extends Application<OntologyConfiguration> {
 
     public static void main(String[] args) throws Exception {
+        String dsn = System.getProperties().getProperty("sentry.dsn");
+        if (null != dsn && !dsn.isEmpty()) {
+            RavenBootstrap.bootstrap(System.getProperties().getProperty("sentry.dsn"));
+            Thread.currentThread().setUncaughtExceptionHandler(UncaughtExceptionHandlers.systemExit());
```
Comment (davidangb) [inline src/main/java/org/broadinstitute/dsde/consent/ontology/OntologyApp.java:34]: do you need to set the system-exit exception handler *before* the bootstrap?

Comment (rushtong) [reply src/main/java/org/broadinstitute/dsde/consent/ontology/OntologyApp.java:34]: According to the plugin authors, Raven bootstrap happens first: https://github.com/tradier/dropwizard-raven
The idea is that we want to send anything that causes a startup error to sentry, and if Raven were not instantiated, nothing would be sent.

Comment (davidangb) [reply src/main/java/org/broadinstitute/dsde/consent/ontology/OntologyApp.java:34]: ahhhh. I see. I thought you were trying to exit on any errors during the Raven bootstrap. All good.

**Review by davidangb — APPROVED** [review]
:+1: lgtm, did not test

**Review by aednichols — APPROVED** [review]

**Pull request merged**